Optimize docker image

pkarakal opened the pull request:
This reduces the number of underlying layers of the docker image, by reducing the multiple `RUN` instructions and making them all one. 
For easier readability I also moved the script to install julia to a separate file. 

Theses changes result in an image reduction of 2GB.

Before: 

```console
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
hpcimage           latest              8458a9716487        6 hours ago         5.41GB
```

After 

```console
REPOSITORY                                       TAG     IMAGE ID      CREATED         SIZE
hpctest                                        latest  77588f69e7f3  5 minutes ago   3.5 GB
```

pkarakal committed:
style: move julia setup script to a separate file

pkarakal committed:
refactor(Dockerfile): Optimize dockerfile

bnase commented:
@pkarakal Something I ve been meaning to do for a while now :) We were adding a lot of ad-hoc things along the way

pkarakal committed:
feat(Dockerfile): Add openblas to dependencies

pkarakal commented:
I updated the dockerfile to include the openblas library in the last commit, if that's ok with you. I can cherry-pick it to another branch if you prefer it that way. (it was a minor change so I included it here)

pkarakal commented:
Ι can rebase and get this updated during the weekend if that works for you.

pitsianis commented:
You are doing us a great favor!

> On Nov 19, 2024, at 9:18 PM, Pavlos Karakalidis ***@***.***> wrote:
> 
> 
> Ι can rebase and get this updated during the weekend if that works for you.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/AUTh-CSAL/Docker/pull/4#issuecomment-2486667064>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAMWJXSWW2RL3H3LS7YFE4L2BOMJRAVCNFSM6AAAAABRY4KVUCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIOBWGY3DOMBWGQ>.
> You are receiving this because you commented.
>

pkarakal committed:
chore: upgrade to latest versions

pkarakal commented:
I rebased the branch on top of the latest main and upgraded all deps to the latest. I also removed the previous script used to install Julia and opted to use the official juliaup version manager. 

I also made the opencilk and Julia versions configurable so you can always just change/specify them during docker build. Let me know if you need any more changes.
